docs/package/rootfs.md

```diff
@@ -0,0 +1,375 @@
+---
+layout: default
+title: "Creating rootfs for unikernels"
+description: "Packaging and creating unikernel's rootfs"
+---
+
+# Packaging and creating unikernel's rootfs for `urunc`
+
+The unikernel and libOS ecosystem is highly diverse, with each framework
+offering its own approach to storage. The users can easily get lost on the
+various storage technologies that each framework supports.  This challenge was
+one of the key reasons we created [bunny](https://github.com/nubificus/bunny) and [bunix](https://github.com/nubificus/bunix), in an effort to simplify the
+process and provide a unified interface for managing storage across all
+unikernel frameworks. On this page, we will explore the current state of our
+tools and explain how to use them to create and package a root filesystem
+(rootfs) for a unikernel.
+
+For the time being, `urunc` supports two ways for passing the rootfs to the
+unikernel: a) through initrd and b) as a virtio-block. In the latter case,
+`urunc` can either levarage the container's snapshot and pass the whole
+container's rootfs as the rootfs, or `urunc` can make use of
+a user-created file inside the OCI image to pass as a virtio-block device
+to the unikernel.
+
+Therefore, the users have the following options:
+
+1. Manually create a rootfs (either initrd or block) and package it along with
+   the unikernel.
+2. Directly copy all the files to the container's rootfs and use devmapper
+   snapshotter, in order to allow `urunc` to pass the snapshot as a virtio-block
+   to the unikernel.
+3. Let [bunny](https://github.com/nubificus/bunny) and [bunix](https://github.com/nubificus/bunix) create the rootfs file.
+
+> **NOTE**: For the time being, [bunny](https://github.com/nubificus/bunny) supports the creation of initrd files and [bunix](https://github.com/nubificus/bunix) does not provide any support for creating the rootfs.
+
+## Creating an initrd file
+
+Some unikernel frameworks and guests support an in-memory ramfs as a rootfs. In
+these cases, we can use [bunny](https://github.com/nubificus/bunny) and instruct it to create the rootfs for
+us with all the specified files.
+This feature is only supported using a `bunnyfile` and not a Dockerfile-like
+syntax file.
+
+Let's take a look at it using a Redis [Unikraft](https://unikraft.org) unikernel as an example, targeting
+[Qemu](https://qemu.org). We will define the `bunnyfile` as:
+
+```
+#syntax=harbor.nbfc.io/nubificus/bunny:0.0.2
+version: v0.1
+
+platforms:
+  framework: unikraft
+  monitor: qemu
+  architecture: x86
+
+rootfs:
+  from: scratch
+  type: initrd
+  include:
+  - redis.conf:/conf/redis.conf
+
+kernel:
+  from: local
+  path: redis-qemu-x86_64-initrd_qemu-x86_64
+
+cmdline: "redis-server /conf/redis.conf"
+```
+
+In the above file we specify the followings:
```

Comment:
I think the correct form would be "following".

Reply:
Sure, let me change that